src/machines/activity_state_machine.rs

```diff
@@ -319,30 +403,47 @@ impl Started {
         // notify_timed_out
         ActivityMachineTransition::default::<TimedOut>()
     }
-    pub(super) fn on_canceled(self) -> ActivityMachineTransition {
-        // createRequestCancelActivityTaskCommand
-        ActivityMachineTransition::default::<Failed>()
+    pub(super) fn on_canceled(self, dat: SharedState) -> ActivityMachineTransition {
+        create_request_cancel_activity_task_command(
+            dat,
+            StartedActivityCancelCommandCreated::default().into(),
+        )
     }
 }
 
 #[derive(Default, Clone)]
 pub(super) struct ScheduledActivityCancelCommandCreated {}
 
 impl ScheduledActivityCancelCommandCreated {
-    pub(super) fn on_command_request_cancel_activity_task(self) -> ActivityMachineTransition {
+    pub(super) fn on_command_request_cancel_activity_task(
+        self,
+        dat: SharedState,
+    ) -> ActivityMachineTransition {
         // notifyCanceledIfTryCancel
-        ActivityMachineTransition::default::<ScheduledActivityCancelCommandCreated>()
+        match dat.cancellation_type {
+            ActivityCancellationType::TryCancel => notify_canceled(
+                dat,
+                None,
+                ScheduledActivityCancelCommandCreated::default().into(),
+            ),
+            _ => ActivityMachineTransition::default::<ScheduledActivityCancelCommandCreated>(),
+        }
     }
 }
 
 #[derive(Default, Clone)]
 pub(super) struct ScheduledActivityCancelEventRecorded {}
 
 impl ScheduledActivityCancelEventRecorded {
-    pub(super) fn on_activity_task_canceled(self) -> ActivityMachineTransition {
+    pub(super) fn on_activity_task_canceled(
+        self,
+        dat: SharedState,
+        _attrs: ActivityTaskCanceledEventAttributes,
+    ) -> ActivityMachineTransition {
         // notify_canceled
```

Comment:
I still think we should just get rid of these little Java reminders. They're not terribly helpful any more

Reply:
Done